sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/SupportsPushDownAggregates.java

```diff
@@ -47,6 +47,13 @@
 @Evolving
 public interface SupportsPushDownAggregates extends ScanBuilder {
 
+  /**
+   * Whether enable aggregation push-down.
+   *
+   * @return true if enable aggregation push-down, false otherwise.
+   */
+  default boolean enablePushAggregation() { return true; }
```

Comment:
Hm, what about just leveraging a configuration for it? Also, users can easily do this via:

```scala
if (!jdbcOptions.pushDownAggregate) return false
```

at `pushAggregation`.

Wonder if it's worth adding another method here.

Reply:
The configuration is different between data source.
`enablePushAggregation` make a chance to skip a lot of conversions.

Reply:
Hm, but this requires each sources to implement too, and I don't think the conversion time can be a major reason to add this because it's trivial compared to actual runtime, etc.

Reply:
Other method looks cheap.

Reply:
Other sources no need to implement it. Because the default value is true. If other sources want keep the origin code, no need any change.

Reply:
I share the same concerns as @HyukjinKwon. It doesn't seem to me that we will have much performance gain here. It's probably not worth it to add another method.

Reply:
The current implement do a lot work(e.g. collect aggregates, normalize expressions, translate aggregation, split AVG into SUM and COUNT and so on). After all these efforts, you told me that the parameter configuration did not agree to push down, which was obviously unreasonable.

Reply:
> The current implement do a lot work(e.g. collect aggregates, normalize expressions, translate aggregation, split AVG into SUM and COUNT and so on).

Have you measured the actual performance penalty? I'd strongly be against adding a new API just for a tiny performance gain.

Reply:
I got it.